### DISCUSSION TITLE
Can I run serverless functions recursively?

### DISCUSSION THREAD

**Discussion by edgarasben** [vercel/community#3464] [Feature Question] [answered]
### Question

I am curious if I am able to run Vercel / Next.js Serverless functions recursively? Meaning that in the end of the function I would like to call the function again, and that way divide the load to multiple functions and avoid the timeout. Something like this:

```ts
import { NextResponse } from 'next/server'

export async function GET(request: Request) {
  const ROOT_URL = process.env.VERCEL_URL ?? 'localhost:3000'

  const { searchParams } = new URL(request.url)
  let timesToRun = parseInt(searchParams.get('timesToRun') ?? '0')

  console.log('timesToRun', timesToRun)

  if (isNaN(timesToRun)) {
    return NextResponse.json({ message: 'timesToRun must be a number' })
  }

  if (timesToRun > 0) {
    const res = await fetch(
      `https://webhook.site/f7d40a5e-3cce-4128-8b94-b12c92e0db1f?timesToRun=${timesToRun}?url=${ROOT_URL}`
    )

    // Delay for 5 seconds before making the next call
    await new Promise((resolve) => setTimeout(resolve, 5000))

    timesToRun--

    fetch(`http://${ROOT_URL}/api/?timesToRun=${timesToRun}`)
  }

  return NextResponse.json('ok')
}

```

This works on localhost but I can also see in my terminal that the function timeout keeps extending after every call, instead of starting from 0:
<img width="907" alt="image" src="https://github.com/vercel/community/assets/725324/2fa0cfae-b1c6-47e0-a270-1508958eed2c">

When I test this in Vercel, setTimeout is ignored, and only one or a few requests are sent out all at once.

**Comment by leerob** [comment, marked as answer]
https://vercel.com/changelog/automatic-recursion-protection-for-vercel-serverless-functions

**Reply by edgarasben** [reply]
I can see how this is useful in many cases, but unfortunately prevents me building a mini queueing system...

I wish Vercel made queuing possible. For example, triggering 20k request over a period of 30mins. Could not find a good and easy solution to this. And the recursive serverless function was my final hope:)
